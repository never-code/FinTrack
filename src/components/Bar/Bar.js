import React, { useEffect } from 'react';
import Highcharts from 'highcharts';

const Bar = ({ data, categories, backgroundColor }) => {
  useEffect(() => {
    Highcharts.chart('basic-bar', {
      chart: {
        type: 'column',
        backgroundColor,
      },
      colors: ['#FBD94F', '#0170F3'],
      title: {
        text: 'Yearly Breakup',
        style: { color: '#0170F3', fontWeight: 900, fontSize: '20px' },
      },
      xAxis: {
        categories,
      },
      yAxis: {
        min: 0,
        title: {
          text: 'Amount',
        },
      },
      tooltip: {
        pointFormat: '<span style="color:{series.color}">{series.name}</span>: <b>{point.y}</b> ({point.percentage:.0f}%)<br/>',
        shared: true,
      },
      plotOptions: {
        column: {
          stacking: 'percent',
        },
      },
      series: data,
    });
  }, [data]);

  return (
    <div id="basic-bar"></div>
  );
};

export default Bar;
