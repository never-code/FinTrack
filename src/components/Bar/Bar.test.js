import React from 'react';
import { render } from '@testing-library/react';
import Bar from './Bar';

const data = [{
  name: 'Test',
  data: [1, 2, 3, 4],
}];
const categories = ['Cat1', 'Cat2'];
describe('Bar Graph: ', () => {
  it('should render successfully', () => {
    const { container } = render(<Bar data={data} categories={categories} backgroundColor="#FFFF"/ >);
    expect(container).toMatchSnapshot();
  });
});
