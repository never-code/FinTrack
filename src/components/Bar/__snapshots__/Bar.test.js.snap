// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar Graph:  should render successfully 1`] = `
<div>
  <div
    data-highcharts-chart="0"
    id="basic-bar"
    style="overflow: hidden;"
  >
    <div
      class="highcharts-container "
      dir="ltr"
      id="highcharts-lx2nao6-0"
      style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none;"
    >
      <svg
        class="highcharts-root"
        height="400"
        style="font-family:\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Arial, Helvetica, sans-serif;font-size:12px;"
        version="1.1"
        viewBox="0 0 600 400"
        width="600"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc>
          Created with Highcharts 8.1.2
        </desc>
        <defs>
          <clippath
            id="highcharts-lx2nao6-1-"
          >
            <rect
              fill="none"
              height="NaN"
              width="NaN"
              x="0"
              y="0"
            />
          </clippath>
        </defs>
        <rect
          class="highcharts-background"
          fill="#FFFF"
          height="400"
          rx="0"
          ry="0"
          width="600"
          x="0"
          y="0"
        />
        <rect
          class="highcharts-plot-background"
          fill="none"
          height="NaN"
          width="NaN"
          x="NaN"
          y="10"
        />
        <g
          class="highcharts-grid highcharts-xaxis-grid"
          data-z-index="1"
        >
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
          />
        </g>
        <g
          class="highcharts-grid highcharts-yaxis-grid"
          data-z-index="1"
        >
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
            stroke="#e6e6e6"
            stroke-width="1"
          />
        </g>
        <rect
          class="highcharts-plot-border"
          data-z-index="1"
          fill="none"
          height="0"
          width="0"
          x="0"
          y="10"
        />
        <g
          class="highcharts-axis highcharts-xaxis"
          data-z-index="2"
        >
          <path
            class="highcharts-axis-line"
            d="M 0 0"
            data-z-index="7"
            fill="none"
            stroke="#ccd6eb"
            stroke-width="1"
          />
        </g>
        <g
          class="highcharts-axis highcharts-yaxis"
          data-z-index="2"
        >
          <text
            class="highcharts-axis-title"
            data-z-index="7"
            style="color:#666666;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0) rotate(270 0 -9999)"
            x="0"
            y="-9999"
          >
            <tspan>
              Amount
            </tspan>
          </text>
          <path
            class="highcharts-axis-line"
            d="M 0 0"
            data-z-index="7"
            fill="none"
          />
        </g>
        <g
          class="highcharts-series-group"
          data-z-index="3"
        >
          <g
            class="highcharts-series highcharts-series-0 highcharts-column-series highcharts-color-0 highcharts-tracker"
            clip-path="url(#highcharts-lx2nao6-1-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(0,10) scale(1 0.001)"
          >
            <rect
              class="highcharts-point highcharts-color-0"
              fill="#FBD94F"
              height="NaN"
              opacity="1"
              stroke="#ffffff"
              stroke-width="1"
              width="NaN"
              x="NaN"
              y="NaN"
            />
            <rect
              class="highcharts-point highcharts-color-0"
              fill="#FBD94F"
              height="NaN"
              opacity="1"
              stroke="#ffffff"
              stroke-width="1"
              width="NaN"
              x="NaN"
              y="NaN"
            />
            <rect
              class="highcharts-point highcharts-color-0"
              fill="#FBD94F"
              height="NaN"
              opacity="1"
              stroke="#ffffff"
              stroke-width="1"
              width="NaN"
              x="NaN"
              y="NaN"
            />
            <rect
              class="highcharts-point highcharts-color-0"
              fill="#FBD94F"
              height="NaN"
              opacity="1"
              stroke="#ffffff"
              stroke-width="1"
              width="NaN"
              x="NaN"
              y="NaN"
            />
          </g>
          <g
            class="highcharts-markers highcharts-series-0 highcharts-column-series highcharts-color-0"
            data-z-index="0.1"
            opacity="1"
            transform="translate(0,10) scale(1 1)"
          />
        </g>
        <text
          class="highcharts-title"
          data-z-index="4"
          style="color:#0170F3;font-size:20px;font-weight:900;fill:#0170F3;"
          text-anchor="middle"
          x="300"
          y="25"
        >
          <tspan>
            Yearly Breakup
          </tspan>
        </text>
        <text
          class="highcharts-subtitle"
          data-z-index="4"
          style="color:#666666;fill:#666666;"
          text-anchor="middle"
          x="300"
          y="10"
        />
        <text
          class="highcharts-caption"
          data-z-index="4"
          style="color:#666666;fill:#666666;"
          text-anchor="start"
          x="10"
          y="397"
        />
        <g
          class="highcharts-legend"
          data-z-index="7"
          transform="translate(300,362)"
        >
          <rect
            class="highcharts-legend-box"
            fill="none"
            rx="0"
            ry="0"
            visibility="visible"
          />
          <g
            data-z-index="1"
          >
            <g>
              <g
                class="highcharts-legend-item highcharts-column-series highcharts-color-0 highcharts-series-0"
                data-z-index="1"
                transform="translate(8,3)"
              >
                <text
                  data-z-index="2"
                  style="color:#333333;cursor:pointer;font-size:12px;font-weight:bold;fill:#333333;"
                  text-anchor="start"
                  x="21"
                  y="15"
                >
                  <tspan>
                    Test
                  </tspan>
                </text>
                <rect
                  class="highcharts-point"
                  data-z-index="3"
                  fill="#FBD94F"
                  height="12"
                  rx="6"
                  ry="6"
                  width="12"
                  x="2"
                  y="4"
                />
              </g>
            </g>
          </g>
        </g>
        <g
          class="highcharts-axis-labels highcharts-xaxis-labels"
          data-z-index="7"
        >
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            Cat1
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            Cat2
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            2
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            3
          </text>
        </g>
        <g
          class="highcharts-axis-labels highcharts-yaxis-labels"
          data-z-index="7"
        >
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            0
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            25
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            50
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            75
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            100
          </text>
        </g>
        <text
          class="highcharts-credits"
          data-z-index="8"
          style="cursor:pointer;color:#999999;font-size:9px;fill:#999999;"
          text-anchor="end"
          x="590"
          y="395"
        >
          Highcharts.com
        </text>
      </svg>
    </div>
  </div>
</div>
`;
